### PR TITLE
fix(intellij): Use node interpreter to run commands

### LINKED ISSUE #404: 🐛 Error 193 in webstorm plugin: %1 is not a valid Win32 application
### Environment information

```block
CLI:
  Version:                      1.2.2
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           windows

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         unset
  JS_RUNTIME_VERSION:           "v20.2.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/9.6.6"

Biome Configuration:
  Status:                       unset

Workspace:
  Open Documents:               0

Discovering running Biome servers...
```

### What happened?

1. Install biome into project
2. Close project
3. Open this project again

It gives this error
```
   Unhandled exception in [CoroutineName(ProjectImpl), StandaloneCoroutine{Cancelling}@3370b515, Dispatchers.Default]
   
   com.intellij.execution.process.ProcessNotCreatedException: Cannot run program "<path to project>\node_modules\.bin\biome" (in directory "<path to project>"): CreateProcess error=193, %1 is not a valid Win32 application
	at com.intellij.execution.configurations.GeneralCommandLine.createProcess(GeneralCommandLine.java:356)
	at com.intellij.execution.process.OSProcessHandler.startProcess(OSProcessHandler.java:84)
	at com.intellij.execution.process.OSProcessHandler.<init>(OSProcessHandler.java:44)
	at com.intellij.execution.process.CapturingProcessHandler.<init>(CapturingProcessHandler.java:20)
	at com.intellij.execution.util.ExecUtil.execAndGetOutput(ExecUtil.kt:89)
	at com.github.biomejs.intellijbiome.BiomeUtils.getBiomeVersion(BiomeUtils.kt:28)
	at com.github.biomejs.intellijbiome.widgets.BiomeWidgetPresentation.getSelectedValue$lambda$0(BiomeWidgetPresentation.kt:23)
	at com.intellij.openapi.progress.impl.CoreProgressManager$1.run(CoreProgressManager.java:252)
	at com.intellij.openapi.progress.impl.CoreProgressManager.startTask(CoreProgressManager.java:428)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.startTask(ProgressManagerImpl.java:115)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcessWithProgressSynchronously$9(CoreProgressManager.java:512)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$new$0(ProgressRunner.java:87)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$4(ProgressRunner.java:251)
	at com.intellij.openapi.progress.ProgressManager.lambda$runProcess$0(ProgressManager.java:71)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:186)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$13(CoreProgressManager.java:604)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:679)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:635)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:603)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:61)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:173)
	at com.intellij.openapi.progress.ProgressManager.runProcess(ProgressManager.java:71)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$5(ProgressRunner.java:251)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$launchTask$18(ProgressRunner.java:465)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$launchTask$19(ProgressRunner.java:477)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:702)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:699)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:699)
	at java.base/java.lang.Thread.run(Thread.java:833)
	Suppressed: kotlinx.coroutines.internal.DiagnosticCoroutineContextException: [CoroutineName(ProjectImpl), StandaloneCoroutine{Cancelled}@3370b515, Dispatchers.Default]
Caused by: java.io.IOException: Cannot run program "<path to project>\node_modules\.bin\biome" (in directory "<path to project>"): CreateProcess error=193, %1 is not a valid Win32 application
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1143)
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1073)
	at com.intellij.execution.configurations.GeneralCommandLine.startProcess(GeneralCommandLine.java:432)
	at com.intellij.execution.configurations.GeneralCommandLine.createProcess(GeneralCommandLine.java:346)
	... 31 more
Caused by: java.io.IOException: CreateProcess error=193, %1 is not a valid Win32 application
	at java.base/java.lang.ProcessImpl.create(Native Method)
	at java.base/java.lang.ProcessImpl.<init>(ProcessImpl.java:499)
	at java.base/java.lang.ProcessImpl.start(ProcessImpl.java:158)
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1110)
	... 34 more
```

### Expected result

Should not give error

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

### PR DESCRIPTION
**Pull request by victor-teles**
## Fixes #404 - explanation

For windows the plugin need to use `node` interpreter to be able to run `biome lsp-proxy` and `biome format`

## Fix for `biome format` for #402 - explanation

The `biome format` command was running  without `cwd`, now it's set to run on projects root

## Test Plan

- [x] UI Testing - Validate list of quickfixes

### CODE REVIEW

File: editors/intellij/src/main/kotlin/com/github/biomejs/intellijbiome/widgets/BiomeWidget.kt [hunk 1]
```diff
@@ -1,16 +1,86 @@
...
+
+    private fun update(text: String?) {
+        ApplicationManager.getApplication()
+            .invokeLater(
+                {
+                    if (project.isDisposed || component == null) {
+                        return@invokeLater
+                    }
+
+                    component!!.text = text
```
Comment (ematipico) [inline editors/intellij/src/main/kotlin/com/github/biomejs/intellijbiome/widgets/BiomeWidget.kt:78]: Isn't there a risk that `component` is `null` here?

File: editors/intellij/src/main/resources/META-INF/plugin.xml [hunk 2]
```diff
@@ -24,7 +24,8 @@
 
     <extensions defaultExtensionNs="com.intellij">
         <notificationGroup id="Biome" displayType="BALLOON"/>
-        <statusBarWidgetFactory implementation="com.github.biomejs.intellijbiome.widgets.BiomeStatusBarWidgetFactory"/>
+        <statusBarWidgetFactory implementation="com.github.biomejs.intellijbiome.widgets.BiomeStatusBarWidgetFactory"
+																id="BiomeWidget"/>
```
Comment (ematipico) [inline editors/intellij/src/main/resources/META-INF/plugin.xml:28]: There's an indentation issue. Too many tabs I think

File: editors/intellij/src/test/testData/basic-project/package.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
+{
+  "name": "basic-project",
+  "version": "1.0.0",
```
Comment (ematipico) [inline editors/intellij/src/test/testData/basic-project/package.json:3]: Please add `"private": true` to the `package.json`

File: editors/intellij/src/test/testData/basic-project/package.json [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "license": "MIT",
+	"dependencies": {
+		"@biomejs/biome": "1.2.2"
```
Comment (ematipico) [inline editors/intellij/src/test/testData/basic-project/package.json:11]: Let's use pnpm workspaces. 
```suggestion
		"@biomejs/biome": "workspace:*"
```

And let's add the `src/test` folder in `pnpm-workspace.yml`.

Like this, we always use the local version. Doing so, would require building the npm package first and, hence, the binaries.

**Review by ematipico — COMMENTED** [review]
Thank you @victor-teles ! I would be nice to have a section in the `CONTRIBUTING.md` that explains a bit the UI testing and how a maintainer can run them locally

File: .github/workflows/release_lsp.yml [hunk 5]
```diff
@@ -273,10 +273,68 @@ jobs:
           fail_on_unmatched_files: true
           generate_release_notes: true
 
+  test-ui-intellij:
```
[the remark this answers is not in the document]
Comment (victor-teles) [reply .github/workflows/release_lsp.yml:276]: I'll gonna add to pull_request.yml workflow

File: .github/workflows/release_lsp.yml [hunk 6]
```diff
@@ -273,10 +273,68 @@ jobs:
...
+    name: Test UI Intellij
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
```
[the remark this answers is not in the document]
Comment (victor-teles) [reply .github/workflows/release_lsp.yml:280]: With fail-fast: false, we could know whether the test fail on a specify OS or it is a generalized failure.
What do you think? it worth it?

**Comment by victor-teles** [comment]
> Thank you @victor-teles ! I would be nice to have a section in the CONTRIBUTING.md that explains a bit the UI testing and how a maintainer can run them locally

@ematipico 
Thank you for the review 

I'll update CONTRIBUTING.md and add the new README.md

File: editors/intellij/src/test/testData/basic-project/package.json (hunk 4 above)
Comment (victor-teles) [reply editors/intellij/src/test/testData/basic-project/package.json:11]: My very initial idea is to test on multiple package managers and maybe with monorepos
For now, I'll put pnpm on and plan the other cases better

File: editors/intellij/src/main/kotlin/com/github/biomejs/intellijbiome/widgets/BiomeWidget.kt (hunk 1 above)
Comment (ematipico) [reply editors/intellij/src/main/kotlin/com/github/biomejs/intellijbiome/widgets/BiomeWidget.kt:78]: The `null` check si together with `project.isDisposed` and it's an `OR` operation, this means that the project can be disposed but the component can still be null

File: .github/workflows/release_lsp.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (victor-teles) [reply .github/workflows/release_lsp.yml:276]: I'll need create a new workflow file to that

File: editors/intellij/src/main/kotlin/com/github/biomejs/intellijbiome/widgets/BiomeWidget.kt (hunk 1 above)
Comment (victor-teles) [reply editors/intellij/src/main/kotlin/com/github/biomejs/intellijbiome/widgets/BiomeWidget.kt:78]: Done! made some code clean up

File: .github/workflows/release_lsp.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (ematipico) [reply .github/workflows/release_lsp.yml]: Let's keep it!

**Review by ematipico — APPROVED** [review]

File: editors/intellij/README.md [hunk 7]
```diff
@@ -1,44 +1,55 @@
-# Biome JetBrains IDEs Plugin
 
-https://biomejs.dev plugin for JetBrains IDEs.
+[Biome](https://biomejs.dev/) is a powerful tool designed to enhance your development experience. This plugin integrates seamlessly with many [JetBrains IDE's](#Supported IDEs) to provide some capabilities:
```
Comment (nissy-dev) [inline editors/intellij/README.md:2]: This is great! the updated readme is really helpful for users!

I think this should be reflected to META-INF/plugin.xml by follow up PR. (We've passed on using `org.jetbrains.changelog.markdownToHTML` in the past, but maybe it's time to use)

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -190,6 +190,16 @@ Build and run the plugin requires:
 ./gradlew buildPlugin
 ```
 
+### UI Testing intellij plugin
+
+Before start testing the plugin you will need to start IDE by invoking the `./gradlew runIdeForUiTests &` and wait IDE startup
```
Comment (nissy-dev) [inline CONTRIBUTING.md:195]: I think it is better to mention `pnpm install` is needed

**Review by nissy-dev — APPROVED** [review]
Thank you for quick fix! I think this PR is ready to merge. 

I commented on some of the points I would like to see addressed in another follow up PR.

**Comment by nissy-dev** [comment]
By the way, the UI test did not work on my end with the following error.
I don't think this needs to be fixed right now, as fixing the bug is the first thing to do.


test result 
```
16:13 🐧 ❯ ./gradlew test
Calculating task graph as configuration cache cannot be reused because the file system entry 'build/classes/kotlin/main/classpath.index' has been created.

> Configure project :
Kotlin DSL property assignment is an incubating feature.

> Task :test
CompileCommand: exclude com/intellij/openapi/vfs/impl/FilePartNodeRoot.trieDescend bool exclude = true

BasicProjectNpmTest > openQuickFixes(RemoteRobot) FAILED
    org.opentest4j.AssertionFailedError at BasicProjectNpmTest.kt:63

1 test completed, 1 failed

> Task :test FAILED

0 problems were found storing the configuration cache.

See the complete report at file:///Users/d002024/code/github.com/biomejs/biome/editors/intellij/build/reports/configuration-cache/517ysd6f9umxr929v0i5rmazs/dwau8wqkvt5e8qfxn26lff4zf/configuration-cache-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///Users/d002024/code/github.com/biomejs/biome/editors/intellij/build/reports/tests/test/index.html

* Try:
> Run with --scan to get full insights.

BUILD FAILED in 19s
12 actionable tasks: 5 executed, 7 up-to-date
Configuration cache entry stored.
```

IDE error

```
2023-09-30 16:13:24,274 [  54437]   WARN - #c.i.o.v.i.l.LocalFileSystemBase - /dev/fd/20: Bad file descriptor
java.nio.file.FileSystemException: /dev/fd/20: Bad file descriptor
        at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:100)
        at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:106)
        at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
        at java.base/sun.nio.fs.UnixFileAttributeViews$Basic.readAttributes(UnixFileAttributeViews.java:55)
        at java.base/sun.nio.fs.UnixFileSystemProvider.readAttributes(UnixFileSystemProvider.java:148)
        at java.base/java.nio.file.Files.readAttributes(Files.java:1851)
        at com.intellij.openapi.util.io.NioFiles.readAttributes(NioFiles.java:177)
        at com.intellij.util.io.PlatformNioHelper.visitDirectory(PlatformNioHelper.java:54)
        at com.intellij.openapi.vfs.impl.local.LocalFileSystemImpl.listWithAttributes(LocalFileSystemImpl.java:307)
        at com.intellij.openapi.vfs.impl.local.LocalFileSystemImpl.lambda$new$0(LocalFileSystemImpl.java:49)
        at com.intellij.openapi.vfs.DiskQueryRelay.accessDiskWithCheckCanceled(DiskQueryRelay.java:45)
        at com.intellij.openapi.vfs.impl.local.LocalFileSystemImpl.listWithCaching(LocalFileSystemImpl.java:274)
        at com.intellij.openapi.vfs.newvfs.RefreshWorker.partialDirRefresh(RefreshWorker.java:321)
        at com.intellij.openapi.vfs.newvfs.RefreshWorker.processQueue(RefreshWorker.java:185)
        at com.intellij.openapi.vfs.newvfs.RefreshWorker.lambda$parallelScan$0(RefreshWorker.java:121)
        at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
```

**Comment by victor-teles** [comment]
@nissy-dev Thank you very much for the review, I will address your points in another PR
I'll let you know when I open the PR

**Pull request merged**